R/fft_graph.R

```diff
@@ -0,0 +1,103 @@
+## FFT Plot Development
+
+# Setup ----
+
+library(dplyr)
+library(ggplot2)
+library(scwplot)
+
+# Detailed Plots ----
+
+## GP Practice ----
+# practice_code and question_number dependent on the selected item on the app
+
+
+fft_practice_graph <- filter(
+  fft_final, practice_code == "K84045"
+)
+
+# gppt_practice_graph <- gppt_practice_graph[order(gppt_practice_graph$sort_order),]
+
+fft_practice_stack <- ggplot(fft_practice_graph, aes(x = month, y = value, fill = reorder(answers, sort_order))) +
+  geom_bar(
+    stat = "identity",
+    position = "stack"
+  ) +
+  geom_hline(yintercept = 0, size = 1, colour = "#333333") +
+  theme_scw() +
+  scale_fill_diverging(discrete = TRUE) +
+  scale_y_continuous(labels = scales::label_number_si()) +
+  labs(
+    title = "Friends and Family Test",
+    subtitle = fft_practice_graph$practice_name,
+    y = "Number of Patients",
+    x = "Month"
+  )
+fft_practice_stack
```

Comment:
I don't think you need to store these plots as objects really. It should be fine to return them directly from the function.

Reply:
Cleaned up some of the code for producing the plots, including calling them directly, rather than storing them as objects.